projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts

```diff
@@ -1,43 +1,147 @@
 import { LogLevel } from '../logging/log-level';
 
 export interface OpenIdConfiguration {
+  /**
+   * The url to the Security Token Service (STS) server.
+   * This field is required.
+   */
   stsServer?: string;
+  /** Override the default STS wellknown endpoint postfix. */
   authWellknownEndpoint?: string;
+  /** The redirect URL defined on the STS. */
   redirectUrl?: string;
+  /**
+   * The Client MUST validate that the aud (audience) Claim contains its client_id value
+   * registered at the Issuer identified by the iss (issuer) Claim as an audience.
+   * The ID Token MUST be rejected if the ID Token does not list the Client as a valid audience,
+   * or if it contains additional audiences not trusted by the Client.
+   */
   clientId?: string;
+  /**
+   * 'code', 'id_token token' or 'id_token' Name of the flow which can be configured.
+   * You must use the 'id_token token' flow, if you want to access an API or get user data from the server.
+   * The access_token is required for this, and only returned with this flow.
+   */
   responseType?: string;
+  /**
+   * This is this scopes which are requested from the server from this client.
+   * This must match the STS server configuration.
+   */
   scope?: string;
+  /**
+   * Optional hd parameter for Google Auth with particular G Suite domain,
+   * see https://developers.google.com/identity/protocols/OpenIDConnect#hd-param
+   */
   hdParam?: string;
+  /** URL after a server logout if using the end session API. */
   postLogoutRedirectUri?: string;
+  /**	Starts the OpenID session management for this client. */
   startCheckSession?: boolean;
+  /** Renews the client tokens, once the token_id expires. Can use the iframes, or the refresh tokens */
   silentRenew?: boolean;
+  /** An optional URL to handle silent renew callbacks */
   silentRenewUrl?: string;
+  /**
+   * Sets the maximum waiting time for silent renew process. If this time is exceeded, the silent renew state will
+   * be reset. Default = 20.
+   * */
   silentRenewTimeoutInSeconds?: number;
+  /**
+   * Makes it possible to add an offset to the silent renew check in seconds.
+   * By entering a value, you can renew the tokens, before the tokens expire.
+   */
   renewTimeBeforeTokenExpiresInSeconds?: number;
+  /**
+   * When set to true, refresh tokens are used to renew the user session.
+   * When set to false, standard silent renew is used.
+   * Default value is false.
+   */
   useRefreshToken?: boolean;
+  /**
+   * Activates Pushed Authorisation Requests for login and popup login.
+   * Not compatible with Iframe renew.
+   */
   usePushedAuthorisationRequests?: boolean;
+  /**
+   * 	A token obtained by using a refresh token normally doesn't contain a nonce value.
+   * 	However, some OIDC endpoint implementations do send one. The library checks to see if the nonce is present.
+   * 	Note that if the nonce value is present, it will not be verified.
+   * 	Default value is false
```

Comment:
append

This is not recommended, if the STS returns a nonce in the refresh.

Reply:
This is now corrected.